### ISSUE TITLE
Attempting to initialize InverseDynamicsSolver with model in MEX function on OpenSim 4.5.1 crashes Matlab

### ISSUE THREAD

**Issue by SpencerTWilliams** [opensim-org/opensim-core#3905]
For the NMSM Pipeline's Treatment Optimization tools, we use the inverse dynamics solver and point kinematics through MEX functions. The functions (C++ source and MEX compiled for Windows attached) are designed so that when one argument is passed (an OpenSim model file name), static variables are initialized to be used in future iterations. 

When initializing the two functions, `pointKinematics` works correctly with one argument, but `inverseDynamicsWithExtraCalcsMexWindows` crashes. The only difference between the `if (nrhs == 1)` blocks in each function is that the inverse dynamics function includes this line:

```c++
idSolver[i] = new InverseDynamicsSolver(*osimModel[i]);
```

These functions both work as expected on version 4.5 of the OpenSim API, so this seems to be a bug introduced in 4.5.1. 

[MEX functions.zip](https://github.com/user-attachments/files/16911567/MEX.functions.zip)

**Comment by nickbianco** [comment]
Hi @SpencerTWilliams, thanks for raising this issue. The `InverseDynamicsSolver` hasn't had any recent changes, so I'm not sure that is the root issue here. Is it possible to run a debugger through `inverseDynamicsWithExtraCalcsMexWindows` to get a stack trace?

**Comment by aymanhab** [comment]
@SpencerTWilliams indeed a stack trace would be key as it's a huge haystack to sort through otherwise.

**Comment by SpencerTWilliams** [comment]
After some more testing, I found that this issue was caused by using a MEX file compiled using resources from the OpenSim 4.4 `sdk` folder. This wasn't an issue when we updated to OpenSim 4.5, but having 4.5.1 linked is not compatible with this older MEX file. Using a new MEX file compiled with the 4.5.1 `sdk` folder works as expected. 

For future reference, this is not a bug with OpenSim, just a potential compatilibity issue for older MEX files. We'll need to check the user's linked OpenSim version in our code to make sure we use appropriate functions.
